app/assets/src/components/Login.jsx

```diff
@@ -66,23 +69,37 @@ class Login extends React.Component {
   }
 
   isFormInValid() {
-    if (this.refs.email.value === "" && this.refs.password.value === "") {
-      this.setState({
-        showFailedLogin: true,
-        errorMessage: "Please enter email and password"
+    const logError = () =>
+      logAnalyticsEvent("Login_login-form-error_displayed", {
+        errorMessage: this.state.errorMessage
       });
+
+    if (this.refs.email.value === "" && this.refs.password.value === "") {
```

Comment:
This could be more compact and without using state... something like
```
errors = compact([
  this.refs.email.value === "" ? "email": null,
  this.refs.password.value === "" ? “Password” : null
]);
if (errors.length > 0) {
  const errorMessage = `Please enter ${errors.join(“ and “)}`;
  this.setState({
    showFailedLogin: true,
    errorMessage
  });
  logError(errorMessage);
  return true;
}
```

Reply:
True, but I'd prefer to keep with purely additive changes to avoid extra manual testing.